### ISSUE TITLE
Error during training: vl_nnloss (line 167) Assertion failed

### ISSUE THREAD

**Issue by paritoshp1412** [vlfeat/matconvnet#872]
Hello all,

I am trying to combine certain feature vectors ( dim: 100x1 ) from different networks using another network which I have defined as follows:

net = dagnn.DagNN();
fc_Block = dagnn.Conv('size', [1 1 101 101], 'pad', [0 0 0 0], 'stride', [1 1]) ;
net.addLayer('alexFeat', fc_Block, {'featA'}, {'alexfc101'}, {'alexfc101_filter', 'alexfc101_bias'});
net.addLayer('googleFeat', fc_Block, {'featG'}, {'googlefc101'}, {'googlefc101_filter', 'googlefc101_bias'});
net.addLayer('resFeat', fc_Block, {'featR'}, {'resfc101'}, {'resfc101_filter', 'resfc101_bias'});
norm_Block = dagnn.BatchNorm('numChannels', 101) ;
net.addLayer('alexNorm', norm_Block, {'alexfc101'}, {'alexFeatx'}, {'alexNorm_f','alexNorm_b','alexNorm_m'});
net.addLayer('googleNorm', norm_Block, {'googlefc101'}, {'googleFeatx'}, {'googleNorm_f','googleNorm_b','googleNorm_m'});
net.addLayer('resNorm', norm_Block, {'resfc101'}, {'resFeatx'}, {'resNorm_f','resNorm_b','resNorm_m'});
net.addLayer('concNorm', dagnn.Concat(), {'alexFeatx', 'googleFeatx', 'resFeatx'}, {'concFeatx'});
net.addLayer('pool', dagnn.Pooling('method', 'average', 'poolSize', [3 1], 'stride', [3 0], 'pad', [0 0 0 0]), {'concFeatx'}, {'pool'}, {});
net.addLayer('fcFinal', fc_Block, {'pool'}, {'fcFinal101'}, {'fcFinal_filter', 'fcFinal_bias'});
net.addLayer('objective', dagnn.Loss('loss', 'softmaxlog'), {'fcFinal101', 'label'}, {'objective'}, {});
net.addLayer('error', dagnn.Loss('loss', 'classerror'), {'fcFinal101','label'}, 'error') ;
net.addLayer('error5', dagnn.Loss('loss', 'topkerror', 'opts', {'topK', 5}), {'fcFinal101','label'}, 'error5') ;
net.rebuild();

the training parameters are: 
trainOpts.batchSize = 100 ;
trainOpts.numEpochs = 5 ;
trainOpts.continue = true ;
trainOpts.gpus = [1] ;
trainOpts.learningRate = 0.0001 ;
trainOpts.weightDecay = 0.0000001 ;
trainOpts.momentum = 0.9 ;

the whole thing works fine during the forward pass but gives an assertion fail error during the backward pass. Seemingly the input to vl_nnconv() becomes a null matrix. 
![image](https://cloud.githubusercontent.com/assets/25795789/22978752/ffa167a2-f3b9-11e6-8be6-b2d28561ae5c.png)

**Comment by layumi** [comment]
@paritoshp1412 
You may forget   trainOpts.derOutputs = {'objective', 1} ;

**Comment by XuYunqiu** [comment]
hi,
I got the same problem

**Comment by layumi** [comment]
Check the input of your network.
Alternatively, I usually set the `opts.conserveMemory = false;` (keep the result of every layer) and run a single forward pass `net.eval(single(rand(224,224,3)))` to see where error appears.
Check the output size of every layer by review `net.vars`. It often helps a lot.

**Comment by h612** [comment]
Hi. You just have to debug your  vl_nnloss() function. Check how the inputs X, c are in size and calculations. Good luck

**Comment by XuYunqiu** [comment]
Thank you
I find my error, it's about the network structure.
It helps a lot
